Lib/pickle.py

```diff
@@ -1742,8 +1758,15 @@ def load_stop(self):
 # Shorthands
 
 def _dump(obj, file, protocol=None, *, fix_imports=True, buffer_callback=None):
-    _Pickler(file, protocol, fix_imports=fix_imports,
-             buffer_callback=buffer_callback).dump(obj)
+    # Adding a check if file represents a file_path(str) instead of an _io.BufferedWriter object
+    if isinstance(file, str):
```

Comment:
I'm not sure we should make this change, but if we did, we'd want to support `os.PathLike` objects here as well as `str`s